third_party/txt/src/txt/paragraph.cc

```diff
@@ -1273,7 +1274,13 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(
       left = SK_ScalarMax;
       right = SK_ScalarMin;
       for (const GlyphPosition& gp : run.positions) {
-        if (gp.code_units.start >= start && gp.code_units.end <= end) {
+        text_length_diff = (gp.code_units.end - gp.code_units.start) - 1;
```

Comment:
Nit: You may want to comment on what the exact reasoning behind the following logic is as it is no longer immediately obvious from the implementation.

Reply:
This may also be better named `glyph_length`, as 'text' is a bit ambiguous in a text shaping library :)